File: roll-your-own-test-framework/typescript/src/TestSummary.ts
import { TestResult } from './TestResult.js';

export class TestSummary {
  private readonly _results: TestResult[] = [];

  get results(): readonly TestResult[] {
    return this._results;
  }

  get run(): number {
    return this._results.length;
  }

  get passed(): number {
    return this._results.filter((r) => r.status === 'PASS').length;
  }

  get failed(): number {
    return this._results.filter((r) => r.status === 'FAIL').length;
  }

  get errors(): number {
    return this._results.filter((r) => r.status === 'ERROR').length;
  }

  add(result: TestResult): void {
    this._results.push(result);
  }
}

File: roll-your-own-test-framework/csharp/src/RollYourOwnTestFramework/TestSummary.cs
namespace RollYourOwnTestFramework;

public class TestSummary
{
    private readonly List<TestResult> _results = new();

    public IReadOnlyList<TestResult> Results => _results;
    public int Run => _results.Count;
    public int Passed => _results.Count(r => r.Status == TestStatus.Pass);
    public int Failed => _results.Count(r => r.Status == TestStatus.Fail);
    public int Errors => _results.Count(r => r.Status == TestStatus.Error);

    public void Add(TestResult result) => _results.Add(result);
}
